### PR TITLE
fix Scrollbar

### PR DESCRIPTION
**Pull request by MixailE**

### CODE REVIEW

**Comment by backmeupplz** [comment]
Seems like you're only using `Measure` for `Scrollbar`, add it to the `Scrollbar` component then. Also, what are you measiring? Name your components less ambiguous. Code is more often read than written.

**Comment by backmeupplz** [comment]
Also, don't forget about the PR description, don't be vague

File: src/components/Measure.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { JSX } from 'preact'
+import { useLayoutEffect, useRef, useState } from 'react'
+
+export default function Measure({
```
Comment (T-Damer) [inline src/components/Measure.tsx:4]: Don't forget to remove the default exported function name

Comment (T-Damer) [reply src/components/Measure.tsx:4]: Let's name it `MeasureContainerHeight` or something like this

File: src/components/Measure.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+}: {
+  defaultHeight?: number
+  className?: string
+  children: (args: { height: number }) => JSX.Element
+}) {
+  const container = useRef<HTMLDivElement>(null)
+  const [height, setHeight] = useState(defaultHeight)
+
+  useLayoutEffect(() => {
+    if (container.current) {
```
Comment (T-Damer) [inline src/components/Measure.tsx:17]: Let's remove curly braces after `if`

File: src/components/Measure.tsx (hunk 1 above)
Comment (T-Damer) [inline src/components/Measure.tsx:4]: Remove the default function name and rename the file, please

File: src/components/Measure.tsx (hunk 2 above)
Comment (T-Damer) [inline src/components/Measure.tsx:17]: Let's remove curly braces after if

File: src/components/Scrollbar.tsx [hunk 3]
```diff
@@ -13,18 +25,34 @@ interface ScrollbarProps {
...
+const scrollContainer = classnames(
+  display('flex'),
+  flexDirection('flex-col'),
+  flexGrow('grow'),
+  position('relative')
+)
+
 export default function ({
   children,
   maxHeight = 350,
```
Comment (T-Damer) [inline src/components/Scrollbar.tsx:37]: Rename this prop to `initHeight` or something like this, it doesn't represent the actual `maxHeight`

**Review by T-Damer — CHANGES_REQUESTED** [review]
A few more

File: package.json [hunk 4]
```diff
@@ -37,6 +37,7 @@
     "fortmatic": "^2.4.0",
     "preact": "^10.7.3",
     "qr-code-styling": "^1.6.0-rc.1",
+    "react-resize-detector": "^7.1.1",
```
Comment (upacyxou) [inline package.json:40]: how much does it weigh in brotli format?

File: src/components/Scrollbar.tsx [hunk 5]
```diff
@@ -1,32 +1,48 @@
...
   const thumbRef = useRef() as MutableRef<HTMLDivElement>
   const [thumbHeight, setThumbHeight] = useState(100)
 
   const { overflows, scrollMaxHeight, isOnTop, isOnBottom } = useIsOverflow(
     wrapRef,
-    maxHeight
+    height
   )
 
+  console.log('height', height, 'scrollMaxHeight', scrollMaxHeight)
```
Comment (upacyxou) [inline src/components/Scrollbar.tsx:44]: do we need this console.log?

File: src/components/Scrollbar.tsx [hunk 6]
```diff
@@ -45,12 +61,13 @@ export default function ({
     const numberOfViews = node.scrollHeight / node.clientHeight
     setThumbHeight(100 / numberOfViews)
   }, [])
+
```
Comment (upacyxou) [inline src/components/Scrollbar.tsx:64]: I guess it could be removed

Comment (T-Damer) [reply src/components/Scrollbar.tsx:64]: Doesn't the code above already listens for resizes?

Comment (MixailE) [reply src/components/Scrollbar.tsx:62]: It listens for changes inside the scroll

File: package.json (hunk 4 above)
Comment (MixailE) [reply package.json:40]: image

**Review by T-Damer — APPROVED** [review]
lgtm

**Pull request merged**